datatypes/mcs_decimal.cpp

```diff
@@ -208,6 +208,21 @@ namespace datatypes
         return toString(const_cast<execplan::IDB_Decimal&>(value));
     }
 
+    // Compare perf with f(string&, ColTypeAlias&, int128_t&)
+    int128_t Decimal::int128FromString(const std::string& value,
```

Comment:
This is a questionable interface approach since I return 16 byte from the function but I believe this should be fine from the performance perspective.